dmoj/executors/filesystem_rules.py

```diff
@@ -0,0 +1,23 @@
+class FilesystemPolicy:
+    def __init__(self, *items):
+        self.items = list(items)
+
+    def __add__(self, other):
+        return FilesystemPolicy(*(self.items + other.items))
+
+    def check(self, path):
```

Comment:
I think `check` should also take an `is_file` parameter. If a rule for `Dir('/foo/bar)` exists but `/foo/bar` is actually a directory, it should be disallowed. We know `is_file` from the syscall that got triggered.

Reply:
How exactly do we know `is_file`? Do I need to check for `O_PATH` or something? What about `stat` calls?